cpu/esp_common/freertos/queue.c

```diff
@@ -204,10 +285,33 @@ BaseType_t IRAM_ATTR _queue_generic_send(QueueHandle_t xQueue,
 
             DEBUG("%s pid=%d queue=%p suspended calling thread\n", __func__,
                   thread_getpid(), xQueue);
+
+#if IS_USED(MODULE_ZTIMER_MSEC)
+            _queue_waiting_thread_t wdt = { .queue = &queue->sending,
+                                            .thread = me,
+                                            .timeout = false };
+            ztimer_t tm = { .callback = _queue_timeout,
+                            .arg = &wdt };
+            if (xTicksToWait < portMAX_DELAY) {
+                ztimer_set(ZTIMER_MSEC, &tm, xTicksToWait * portTICK_PERIOD_MS);
+            }
+#else
+            assert((xTicksToWait == 0) || (xTicksToWait == portMAX_DELAY));
```

Comment:
Hm, the ESP implementations should pull in `ztimer` only when network devices are used. The problem is that for some reason the module `libc_gettimeofday` is pulled in anyway, which in turn pulls in `xtimer` and `ztimer_usec`, which seems to be counterproductive regarding the intention of `ztimer_msec` and the effort we spent to change ESP ports so that they only use `ztimer_msec`.

Anyway, from ESP implementation point of view, `ztimer_msec` isn't  used in any case. If it isn't pulled in, time limits can't be used and most ESP-IDF functions don't use time limits. That's why it is a conqditional depending on `ztimer_msec`.

Reply:
The time limit is only used in functions related to `esp_wifi` where `ztimer_msec` is pulled in.

Reply:
The dependency on module `libc_gettimeofday` is removed by PR #18202 so that `ztimer*` is only pulled in if really needed.